app/views/course/assessment/question/text_responses/_text_response.json.jbuilder

```diff
@@ -1,10 +1,32 @@
 json.allowAttachment question.allow_attachment? unless question.hide_text?
+json.comprehension question.comprehension_question?
 json.autogradable question.auto_gradable?
 
 json.solutions question.solutions.each do |solution|
+  json.id solution.id
   json.solutionType solution.solution_type
   # Do not sanitize the solution here to prevent double sanitization.
   # Sanitization will be handled automatically by the React frontend.
   json.solution solution.solution
   json.grade solution.grade
-end if can_grade && question.auto_gradable?
+end if can_grade && question.auto_gradable? && !question.comprehension_question?
+
+json.groups question.groups.each do |group|
```

Comment:
Shall we move this into a separate partial and render if it is a `comprehension_question?`.

Reply:
I'd think it's okay to leave it together...? Currently it only renders `if can_grade && question.auto_gradable? && question.comprehension_question?`.